### PR TITLE
Not to send body when stack-trace is empty

### PR DESCRIPTION
**Pull request by nowsprinting**
### Changes
- Fix not to send body when stack-trace is empty
- Remove suppress diagnostic for obsolete fields
- Remove `AutopilotSettings` from parameter of `PostReportAsync`

---

### Contribution License Agreement

- [x] By placing an "x" in the box, I hereby understand, accept and agree to be bound by the terms and conditions of the [Contribution License Agreement](https://dena.github.io/cla/).

### CODE REVIEW

File: Runtime/Reporters/SlackMessageSender.cs [hunk 1]
```diff
@@ -107,16 +107,20 @@ public SlackMessageSender(SlackAPI api)
...
                 );
                 if (!postScreenshotTask.Success)
                 {
                     return;
                 }
             }
 
-            var body = Body(logString, stackTrace);
-            await _slackAPI.Post(slackToken, slackChannel, body, postTitleTask.Ts);
+            if (stackTrace != null && stackTrace.Length > 0)
```
Comment (Kuniwak) [inline Runtime/Reporters/SlackMessageSender.cs:119]: @nowsprinting Sorry for my late reply. I’m wondering if it’s better to send it as proposed or to use the default message. Could you please explain why you chose not to use the default message?

**Review by Kuniwak — COMMENTED** [review]
I only made one comment. I had no other concerns.

File: Runtime/Reporters/SlackMessageSender.cs (hunk 1 above)
Comment (nowsprinting) [reply Runtime/Reporters/SlackMessageSender.cs:119]: This post is 2nd or 3rd post.
1st post: `logString`
2nd post: screenshot if flag is true

Before this PR, 3rd post body is equivalent to 1st post if `stachTrace` is empty.

Comment (Kuniwak) [reply Runtime/Reporters/SlackMessageSender.cs:119]: Ah, I understood.

**Review by Kuniwak — APPROVED** [review]
LGTM!

**Comment by nowsprinting** [comment]
Thanks!

**Pull request merged**